toolchain/check/deduce.cpp

```diff
@@ -23,10 +23,11 @@ namespace {
 // methods to add and pop pending deductions from the list. Deductions are
 // popped in order from most- to least-recently pushed, with the intent that
 // they are visited in depth-first order, although the order is not expected to
-// matter except when it influences which error is diagnosed.
+// matter except when it influences which error is diagnosed. `context` must not
+// be null.
```

Comment:
This should probably be a constructor comment, not a class comment.

Reply:
Done.